cpp/src/rolling/detail/rolling.hpp

```diff
@@ -35,26 +35,24 @@ struct rolling_store_output_functor {
 // Specialization for MEAN
 template <typename _T>
 struct rolling_store_output_functor<_T, true> {
-  // SFINAE for non-bool types
+  // We need to avoid storing if count is zero because for mean aggregations
+  // this could perform integer division by zero (undefined behaviour), which
+  // could cause the compiler to deduce nonsense about the loop that increments
+  // count.
```

Comment:
Here, we know we're specialized for `mean`. So perhaps just:

> Don't store the output if count is zero since integral division by zero is undefined behaviour. The caller must ensure that the relevant row is marked as invalid with a null.